fix #72491: text properties in lines added by double click

MarcSabatella opened the pull request:
Two separate issues:

1) Cloning a textline from the palette copies the text properties as is, rather than applying the score text style.  Thus the initial style when applying a textline by double click was not honoring the score style.  Fixed by rebasing the style in the same manner that is done when adding text elements (staff text, etc) directly.

2) setScore() on a textline was not setting the score for the text elements, so the text elements within the textline have their _score still set to gscore, which is why reset to style was not working either.  Fixed by defining TextLine::setScore() override.

lasconic commented:
Thanks for changing the if.
Regarding 1/, an additional remark. This method is also called if a user uses Shift + Ctrl + Drag to clone an ottava for example (yes, I know Shift + Ctrl + Drag is not the most documented or tested feature :-/).
In this case, I guess we don't want to reset to style but keep any changes done to the text properties. So we might want to check if the drop is "from palette" or not.

MarcSabatella commented:
The code I changed for 1) is in a function I added recently to handle palette double click specifically, and it is not called anywhere else.  Shift+Ctrl+drag uses the "other" cmdAddSpanner(), the one used for drop oeprations as opposed to double click.

BTW, I had noticed Shift+Ctrl worked to clone a spanner but never knew if this was intended to be a "feature" or just a side effect of the code to support palette customization.  Every so often people ask and I never think to suggest this.

lasconic commented:
Ok. Thanks for the clarification.
As far as I know, Shift+Ctrl+drag is intended for cloning elements for palette or score.
